solidity/test/bridge/Bridge.Wallets.test.ts

```diff
@@ -705,4 +716,773 @@ describe("Bridge - Wallets", () => {
+                ecdsaWalletTestData.publicKeyY
+              )
+          ).to.be.revertedWith("Caller is not the ECDSA Wallet Registry")
+        })
+      })
+    })
+  })
+
+  describe("notifyCloseableWallet", () => {
+    context("when the reported wallet is not the active one", () => {
+      context("when wallet is in Live state", () => {
+        before(async () => {
+          await createSnapshot()
+
+          await bridge.setWallet(ecdsaWalletTestData.pubKeyHash160, {
+            ecdsaWalletID: ecdsaWalletTestData.walletID,
+            mainUtxoHash: ethers.constants.HashZero,
+            pendingRedemptionsValue: 0,
+            createdAt: await lastBlockTime(),
+            moveFundsRequestedAt: 0,
+            state: walletState.Live,
+          })
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        context("when wallet reached the maximum age", () => {
+          before(async () => {
+            await createSnapshot()
+
+            await increaseTime((await bridge.getWalletsParameters()).maxAge)
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          context("when wallet balance is zero", () => {
+            let tx: ContractTransaction
+
+            before(async () => {
+              await createSnapshot()
+
+              tx = await bridge
+                .connect(walletRegistry.wallet)
+                .notifyCloseableWallet(
+                  ecdsaWalletTestData.pubKeyHash160,
+                  NO_MAIN_UTXO
+                )
+            })
+
+            after(async () => {
+              walletRegistry.closeWallet.reset()
+
+              await restoreSnapshot()
+            })
+
+            it("should change wallet's state to Closed", async () => {
+              const { state } = await bridge.getWallet(
+                ecdsaWalletTestData.pubKeyHash160
+              )
+
+              expect(state).to.be.equal(walletState.Closed)
+            })
+
+            it("should emit WalletClosed event", async () => {
+              await expect(tx)
+                .to.emit(bridge, "WalletClosed")
+                .withArgs(
+                  ecdsaWalletTestData.walletID,
+                  ecdsaWalletTestData.pubKeyHash160
+                )
+            })
+
+            it("should call ECDSA Wallet Registry's closeWallet function", async () => {
+              // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+              expect(walletRegistry.closeWallet).to.have.been.calledOnceWith(
+                ecdsaWalletTestData.walletID
+              )
+            })
+          })
+
+          context("when wallet balance is greater than zero", () => {
+            const walletMainUtxo = {
+              txHash:
+                "0xc9e58780c6c289c25ae1fe293f85a4db4d0af4f305172f2a1868ddd917458bdf",
+              txOutputIndex: 0,
+              txOutputValue: 1,
```

Comment:
Is it possible to send 1 sat on Bitcoin? I wonder if we shouldn't have some sort of dust threshold for moving funds functionality below which, we just allow to close the wallet with the dust remaining.

Reply:
Yep, it is possible. You can construct a transaction with multiple outputs that are used to fund the mining fee and add one output with 1 satoshi (or even 0 satoshis). Regarding the dust threshold, this is worth exploring but note that we are not able to easily check the balance in all scenarios that trigger the moving funds process.

Reply:
Captured in https://github.com/keep-network/tbtc-v2/issues/180.